src/Arcus.Messaging.Tests.Integration/ServiceBus/ServiceBusClient.cs

```diff
@@ -33,62 +35,108 @@ public ServiceBusClient(KeyRotationConfig configuration, ITestOutputHelper logge
             _logger = logger;
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public async Task<AccessKeys> GetConnectionStringKeysAsync(ServiceBusEntity entity)
+        {
+            using IServiceBusManagementClient client = await CreateServiceManagementClientAsync();
+
+            return entity switch
+            {
+                ServiceBusEntity.Queue => await client.Queues.ListKeysAsync(
+                    _configuration.ServiceBusNamespace.ResourceGroup,
+                    _configuration.ServiceBusNamespace.Namespace,
+                    _configuration.ServiceBusNamespace.TopicName,
+                    _configuration.ServiceBusNamespace.AuthorizationRuleName),
+                ServiceBusEntity.Topic => await client.Topics.ListKeysAsync(
+                    _configuration.ServiceBusNamespace.ResourceGroup,
+                    _configuration.ServiceBusNamespace.Namespace,
+                    _configuration.ServiceBusNamespace.TopicName,
```

Comment:
What about getting the key for a subscription on a topic? That's how it's done on the receiver side

Reply:
Hmm, I don't think we can do this bc the access keys we want to retrieve doesn't reference the subscription in any way. We can only list it for the current resource, not for any subscriptions.

Reply:
You mean in the SDK or Arcus?

Reply:
I mean the SDK. The way we can pull information about the Azure Service Bus SDK isn't designed in that way.

Reply:
You are correct, my bad! There is no key on a topic subscription level.